app/models/user.rb

```diff
@@ -14,6 +14,19 @@ class User < ApplicationRecord
   validates :email, presence: true, uniqueness: true
   validates :id_number, presence: true, uniqueness: true, numericality: { only_integer: true }, length: { is: 11 }
   validates_with EmailAddress::ActiveRecordValidator, field: :email
+
+  # background jobs
+  after_commit :build_address_information, on: :create, if: proc { addresses.formal.empty? }
+  after_commit :build_identity_information, on: :create, if: proc { identities.formal.empty? }
```

Comment:
Bunun ilk hali `after_create`'ti. Sonra sidekiq'i takip ettiğimde isteklerin ilk seferinde hata aldığını (x ID'li y'yi bulamadım diyor), ikinci denemede geçtiğini gördüm. İşin hikayesi şu, sidekiq o kadar hızlı çalışıyor ki Rails create'i tetikleyip postgres'e kayıt edeceği datayı yolluyor - ama postgres daha datayı kaydetmeden sidekiq istek atıyor ve x ID'sini bulamıyor. after_commit ise 'database'e kayıt yapıldıktan sonra'yı garantiliyor. Sinsi bir bug potansiyeli taşıyan bir mevzu. Tavsiye okuma: https://www.justinweiss.com/articles/a-couple-callback-gotchas-and-a-rails-5-fix/

Reply:
Burada `after_create_commit`' kullansan daha güzel olur. [bkz](https://blog.bigbinary.com/2016/05/11/rails-5-adds-after_create-aliases.html)